authoring-recipes/multiple-versions.md

```diff
@@ -0,0 +1,50 @@
+
+However, what if you're making migration recipes that help others get to the latest version? Your project might need many different versions of the same library. For instance, a Mockito migration project might need Mockito versions 1, 2, 3, and 4.
+
+If you tried to add many versions of the same library in your `build.gradle` or `pom.xml` file, you would find that, by default, only the newest one is used. Without multiple versions, though, you wouldn't be able to test that your recipes were performing correctly as the expected code might not even compile.
+
+Fortunately, OpenRewrite has added some functionality to help with this predicament. Below we'll walk through how to add multiple versions of a library to a project and how to use those different versions in different tests.
+
+## Manually copying jars and using the `classpathFromResources` function
+
+In your project, you can put many different versions (jars) of the same library in the `src/main/resources/META-INF/rewrite` directory. Then, in your tests, you can specify which one(s) you want to use by utilizing the `classpathFromResources` function.
+
+For instance, if you wanted to default to using `junit-4.13.2` and `mockito-core-3.12.4` in your tests, you could add this to your test file:
+
+```java
+@Override
+public void defaults(RecipeSpec spec) {
+    spec
+        .parser(JavaParser.fromJavaVersion()
+            .classpathFromResources("junit-4.13.2", "mockito-core-3.12.4"));
+}
+```
+
+If you wanted to use a different version of these libraries in another test or in another file, all you would need to do is add the corresponding jar to the `src/main/resources/META-INF/rewrite` directory and specify the new version in the `classpathFromResources` function.
+
+### OpenRewrite recipe library
+
+The above solution works and is what you'll need to do for Maven projects. For Gradle projects, however, there's an easier option than manually downloading and copying the jars into the `src/main/resources/META-INF/rewrite` directory.
+
+You can, instead, use the OpenRewrite [recipe-library plugin](https://github.com/openrewrite/rewrite-build-gradle-plugin/blob/main/src/main/java/org/openrewrite/gradle/RewriteRecipeLibraryPlugin.java). You can apply this plugin by adding this to your `build.gradle` file:
+
+```groovy
+plugins {
+    id("org.openrewrite.build.recipe-library") version "latest.release"
+}
+```
+
+With the `recipe-library` plugin added, you can then add a `recipeDependencies` section in your `build.gradle` file. In there, you can add a `parserClasspath` that specifies a dependency and its corresponding version:
+
+```groovy
+recipeDependencies {
+    parserClasspath("org.mockito:mockito-all:1.10.19")
+    parserClasspath("org.mockito:mockito-core:3.+")
+    parserClasspath("org.mockito:mockito-junit-jupiter:3.+")
```

Comment:
I'd maybe keep this in line with the `.classpathFromResources("junit-4.13.2", "mockito-core-3.12.4"));` above, and maybe use two different versions of the same artifact, with the same pinned versions. That makes it a bit more clear this is intended for different versions of the same artifact that would otherwise conflict. Not using version ranges here might help prevent multiple versions from the same range ending up in a repository.

But open to alternative views on all of this!

Reply:
Good suggestion. 

One question, though: What do you mean by "two different versions of the same artifact, with the same pinned versions?"

Reply:
Also - does the new commit help resolve your concerns or did I miss something on that front?